awx/sso/backends.py

```diff
@@ -54,6 +55,19 @@ def __init__(self, prefix='AUTH_LDAP_', defaults={}):
             options[ldap.OPT_NETWORK_TIMEOUT] = 30
             self.CONNECTION_OPTIONS = options
 
+        # when specifying `.set_option()` calls for TLS in python-ldap, the
+        # *order* in which you invoke them *matters*, particularly in Python3,
+        # where dictionary insertion order is persisted
+        #
+        # specifically, it is *critical* that `ldap.OPT_X_TLS_NEWCTX` be set *last*
+        # this manual sorting puts `OPT_X_TLS_NEWCTX` *after* other TLS-related
+        # options (because their openldap keys are < `ldap.OPT_X_TLS_NEWCTX`
+        #
+        # see: https://github.com/python-ldap/python-ldap/issues/55
+        self.CONNECTION_OPTIONS = OrderedDict(
+            (k, v) for k, v in sorted(self.CONNECTION_OPTIONS.items())
```

Comment:
It looks like there are at least a couple of `OPT_X_TLS..` options that come after `OPT_X_TLS_NEWCTX` ([here's the list](https://gist.github.com/jladdjr/b20acfea5f27cbcd5a88e6e40304510b#file-gistfile1-txt-L20) I get when looking at python-ldap). These would get put after `OPT_X_TLS_NEWCTX` if we apply the above sort.

Should we form an OrderedDict then explicitly append `OPT_X_TLS_NEWCTX` to the end? Something like this maybe:

```suggestion
newctx_option = self.CONNECTION_OPTIONS.pop(ldap.OPT_X_TLS_NEWCTX, None)
self.CONNECTION_OPTIONS = OrderedDict(self.CONNECTION_OPTIONS)
if newctx_option:
    self.CONNECTION_OPTIONS[dap.OPT_X_TLS_NEWCTX] = newctx_option
```